### DISCUSSION TITLE
Please reconsider this code in DownloadHelper

### DISCUSSION THREAD

**Discussion by gutley** [mozilla-mobile/firefox-ios#14650] [General]
Recently, a customer of ours has been complaining that on a specific website they are unable to download files. Clicking the download link ends up with nothing happening. Unfortunately the site is behind a login so I can't give you an offending url.

But I have tracked the problem down to this code in DownloadHelper.init:

```swift
        let mimeType = response.mimeType ?? MIMEType.OctetStream
        let isAttachment = mimeType == MIMEType.OctetStream

        // Bug 1474339 - Don't auto-download files served with 'Content-Disposition: attachment'
        // Leaving this here for now, but commented out. Checking this HTTP header is
        // what Desktop does should we ever decide to change our minds on this.
        // let contentDisposition = (response as? HTTPURLResponse)?.allHeaderFields["Content-Disposition"] as? String
        // let isAttachment = contentDisposition?.starts(with: "attachment") ?? (mimeType == MIMEType.OctetStream)

        guard isAttachment || !canShowInWebView || forceDownload else { return nil }
```

By restoring the commented out code, the site now works as expected and the user is asked if they want to download the file.

Looking at the referenced bug [1474339](https://bugzilla.mozilla.org/show_bug.cgi?id=1474339) we see this as the reason for this change:

_To provide a better end-user experience on mobile, we should try to more closely resemble Safari's behavior and not automatically trigger a download even when files are served up with the `Content-Disposition: attachment` HTTP header._

However, things have moved on in the 5 years since this change was made and Safari does now trigger a download for `Content-Disposition: attachment`. So does Chrome.

Please can we reconsider whether this code should be reverted.

**Comment by lmarceau** [comment]
Hello @gutley ! Do you want to open this as a GitHub ticket? Could be a "feature-request". Thank you!
